src/connectors/wire/WireServer.cpp

```diff
@@ -18,6 +18,17 @@ void SocketServer::listen(const port_type port) {
     acceptor.listen(1);
 }
 
+SocketServer::port_type SocketServer::listenPort() const {
+  boost::system::error_code ec;
+  const tcp::endpoint ep(acceptor.local_endpoint(ec));
+  if (ec)
+  {
+    return 0;
+  }
```

Comment:
I don't think any part of the software can handle a failure here (return value of 0). Isn't it better to use the other signature that throws boost::system::system_error to make it crash if for some reason the port number cannot be retrieved?

If you agree, remember to change the documentation for the interface as well :-)

P.S. Position of `{` not consistent with the rest of the code